### ISSUE TITLE
Reminders for todo.txt

### ISSUE THREAD

**Issue by gsantner** [gsantner/markor#716]
#### General information

* **App version:** v2.1.x
* **System:** v9

#### Description
todo.txt format has support for custom key-value pairs. One common supported custom key-value pair is the `due:2019-12-25` tag for specifing the due date of a todo.
The todo.txt implementation at Markor (Stt) already supports and does parse custom tags.

This issue is about making use of the `Map<String,String>` key value pairs that are already extracted. Out of that information with todo`s due dates, schedule notifications for the day of of due.
When opening a file with scheduled notifications, and a due date was removed - also remove the notification.

So TL;DR: Create reminders for todos.

**Comment by jakub231** [comment]
Hello! I would like to work on this issue. Can you assign it on me?

**Comment by gsantner** [comment]
@stefanfris 
Thanks, sure go for it!

The files required to work on are in format/todotxt folder and in opoc/format/todotxt/Stt*.java.

**Comment by gsantner** [comment]
@jakub231
Thanks, sure go for it!

The files required to work on are in format/todotxt folder and in opoc/format/todotxt/Stt*.java.

Please make a pull request as soon as you start, that makes it easier to detect bigger issues and you don't do certain things multiple times :).

**Comment by gsantner** [comment]
@jakub231  Still want to work on this?

**Comment by gsantner** [comment]
Hidden/Removed some comments, so far nobody has started on this.

**Comment by 0815horsti** [comment]
I'm sorry that I'm not a programmer but I have a little comment on this issue:

It would be very helpful if there is a setting in the preferences (prefrences - todo.txt - xxx) where you can set the time, when every notification of that day "fires". If it is set to 09:00 o'clock every due time set will remind you at that point of time.

Greetings

**Comment by Zoo-M0** [comment]
I would like to work on this to improve my knowledge on the theme. Can I take it? Or do you have more important things where I can help?

**Comment by gsantner** [comment]
You do what you like :-).

**Comment by ofossum** [comment]
I'm not a developer BUT i'd like to say, that this feature would be a real killer!
I am looking into different todo-solutions, which are FOSS, cross-platform and on F-Droid for android. 
There is literally none as comfy and capable as todo.txt and working like a charm like markor. The only one which is _only_ focused on todo.txt _and_ has reminders (via calendar) seems to be Simpletask (also on F-Droid).
So if markor could somehow trigger reminders, which could be deferrable etc. would be really great.
Thanks for all the efforts you put into this editor! :)

**Comment by wolfsolver** [comment]
In addition to notification for date, I think that is very usefull to have context identification and notification based on situations. I try to explain my thoughts with a simple example:

- suppose that I have a to-do.txt with a context "@car"
- in setting I will define that "@car" is binded to presence of BT connection with my car
- so suddenly I enter in the car the app can recognize the context "CAR" and show me a reminder for all task connected to context car

I immagine this kind of automated context:
- "@car" ==> BT connection with specific device 
- "@home"  or @Work ==> wifi or gps position
- @grocery can be GPS position 

So notifications can be raised base on date or event
